A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,475 @@
+We distinguish and exclude gRPC GCP observability from this design: gRPC GCP 
+observability is about exporting data while this design is about instrumentation. 
+Migrating gRPC GCP observability to OpenTelemetry is a future project that depends
+on this work.
+
+### Related Proposals and Documents: 
+* [gRFC L29: C++ API Changes for OpenCensus Integration][L29]
+* [gRFC A45: Exposing OpenCensus Metrics and Tracing for gRPC retry][A45]
+* [Microservices observability overview][grpc-observability-public-doc]
+* [gRFC A66: OpenTelemetry Metrics][A66]
+
+## Proposal
+gRPC users depending on the grpc-census plugin have non-trivial migration paths 
+to OpenTelemetry. Consider the following use cases:
+1. Migrate an application binary where both OpenCensus and OpenTelemetry maybe exist
+in the dependency tree. This can be the application’s own tracing code, or gRPC 
+OpenCensus, or other dependencies that involve OpenCensus and/or OpenTelemetry.
+2. Compatibility between a gRPC client and server as two distributed components, 
+where during migration one will use OpenCensus and the other will use OpenTelemetry.
+
+All related APIs here are experimental until OpenTelemetry metrics [gRPC A66][A66] 
+is design and implementation complete.
+
+### New Function In OpenTelemetry Plugin
+We will add tracing functions in grpc-open-telemetry plugin, along with OpenTelemetry 
+metrics [gRPC A66][A66].
+Languages will keep using the same gRPC infrastructures, e.g. interceptor and 
+stream tracer to implement the feature.
+We keep the grpc-census plugin to allow users who already depend on grpc-census to
+continue using it for newer grpc versions.
+In the new function we will produce the same tracing information as we produce 
+for Census. But due to API differences between OpenCensus and OpenTelemetry, the
+trace information is represented slightly differently.
+The OpenCensus `MessageEvent` fields maps to OpenTelemetry event attributes:
+
+| OpenCensus Trace Message Event Fields | OpenTelemetry Trace Event Attribute Key |
+|---------------------------------------|-----------------------------------------|
+| Type                                  | `message.event.type`                    |
+| Message Id                            | `message.message.id`                    |
+| Uncompressed message size             | `message.event.size.uncompressed`       |
+| Compressed message size               | `message.event.size.compressed`         |
+
+OpenCensus span annotation description maps to OpenTelemetry event name, and 
+annotation attributes keys are mapped to event attributes keys:
+
+| OpenCensus Trace Annotation Attribute Key | OpenTelemetry Trace Event Attribute Key |
+|-------------------------------------------|-----------------------------------------|
+| `id`                                      | `message.event.type`                    |
+| `type`                                    | `message.message.id`                    |
```

Comment:
I think the attribute key is mixed up here :).

Reply:
oh i see, thanks for catching it!